src/halmos/sevm.py

```diff
@@ -373,22 +371,61 @@ def __init__(self, **kwargs) -> None:
 class Contract:
     """Abstraction over contract bytecode. Can include concrete and symbolic elements."""
 
+    _code: ByteVec
+    _fastcode: Optional[bytes]
+    _insn: Dict[int, Instruction]
+    _next_pc: Dict[int, int]
```

Comment:
bit of a lateral move, but I didn't find it super elegant to have `next_pc` be part of `Instruction`. Maybe makes more sense to have it be an explicit cache layer in Contract

Reply:
the advantage is that you can build that cache layer while parsing jumpdests but without having to decode instructions